### PR TITLE
Build improvements, UI-improvements

### PR DESCRIPTION
**Pull request by jmlagace**
Made the following tweaks:

- make the docker build faster (especially when rebuilding locally multiple times) by taking advantage of the docker layer caching mechanism
- added a copy button to copy the registry name and tag
- reverse sorting the tags so date and version base tags show at the top of the repository page

### CODE REVIEW

File: app/views/tags/show.html.erb [hunk 1]
```diff
@@ -9,6 +9,10 @@
 
   <table class="table">
     <tr>
+        <th>Docker Pull Command</th>
```
[the remark this answers is not in the document]
Comment (jmlagace) [reply app/views/tags/show.html.erb]: It makes complete sense. Will implement as suggested.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  
+  def get_pull_command(repository, tag = "latest")
+    return "docker pull " + [Rails.configuration.x.public_registry_url, repository].join("/") + ":" + tag
+  end
```
[the remark this answers is not in the document]
Comment (jmlagace) [reply app/helpers/application_helper.rb]: Makes complete sense as well (been doing too much PHP lately apparently).

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+
```
Comment (klausmeyer) [inline app/helpers/application_helper.rb]: ✂️

File: config/initializers/backend.rb [hunk 4]
```diff
@@ -4,4 +4,5 @@
   config.x.basic_auth_user     = Config.get(name: "BASIC_AUTH_USER", secret: true)
   config.x.basic_auth_password = Config.get(name: "BASIC_AUTH_PASSWORD", secret: true)
   config.x.delete_enabled      = Config.get(name: "ENABLE_DELETE_IMAGES").in? %w(1 true yes)
+  config.x.public_registry_url = Config.get(name: "DOCKER_PUBLIC_REGISTRY_URL")
```
Comment (klausmeyer) [inline config/initializers/backend.rb]:
```diff
- Config.get(name: "DOCKER_PUBLIC_REGISTRY_URL")
+ Config.get(name: "PUBLIC_REGISTRY_URL")
```

File: app/views/repositories/show.html.erb [hunk 5]
```diff
@@ -5,13 +5,21 @@
...
-        <%= link_to tag_path(repo: @repository.name, tag: tag), class: "list-group-item" do %>
+    <ul class="list-group">
+      <% @repository.tags.sort.reverse.each do |tag| %>
+        <li class="list-group-item">
+        <%= link_to tag_path(repo: @repository.name, tag: tag) do %>
           <strong>Tag:</strong> <%= tag %>
         <% end %>
+        <% if Rails.configuration.x.public_registry_url %>
+          <div class="pull-right">
+            <span id="pull-command-<%= tag.object_id %>"><%= get_pull_command(@repository.name, tag)%></span>
```
Comment (klausmeyer) [inline app/views/repositories/show.html.erb]: Looks like you forgot to change the method name here:
`get_pull_command` -> `pull_command`.

File: app/views/tags/show.html.erb [hunk 6]
```diff
@@ -8,6 +8,12 @@
   </div>
 
   <table class="table">
+    <% if Rails.configuration.x.public_registry_url %>
+      <tr>
+        <th>Docker Pull Command</th>
+        <td><span id="pull-command"><%= get_pull_command(@repository.name, @tag.name)%></span> <button class="btn btn-default btn-xs glyphicon glyphicon-copy" type="button" onclick="copyToClipboard('#pull-command')"></button></td>
```
Comment (klausmeyer) [inline app/views/tags/show.html.erb]: Same as in prev. comment:
`get_pull_command` -> `pull_command`.

**Pull request merged**